homeassistant/components/pulseaudio_loopback/switch.py

```diff
@@ -58,97 +39,58 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
-        self._pa_host = host
-        self._pa_port = int(port)
-        self._buffer_size = int(buff_sz)
-        self._tcp_timeout = int(tcp_timeout)
-
-    def _send_command(self, cmd, response_expected):
-        """Send a command to the pa server using a socket."""
-        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-        sock.settimeout(self._tcp_timeout)
-        try:
-            sock.connect((self._pa_host, self._pa_port))
-            _LOGGER.info("Calling pulseaudio: %s", cmd)
-            sock.send((cmd + "\n").encode("utf-8"))
-            if response_expected:
-                return_data = self._get_full_response(sock)
-                _LOGGER.debug("Data received from pulseaudio: %s", return_data)
-            else:
-                return_data = ""
-        finally:
-            sock.close()
-        return return_data
-
-    def _get_full_response(self, sock):
-        """Get the full response back from pulseaudio."""
-        result = ""
-        rcv_buffer = sock.recv(self._buffer_size)
-        result += rcv_buffer.decode("utf-8")
-
-        while len(rcv_buffer) == self._buffer_size:
-            rcv_buffer = sock.recv(self._buffer_size)
-            result += rcv_buffer.decode("utf-8")
-
-        return result
-
-    @util.Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
-    def update_module_state(self):
-        """Refresh state in case an alternate process modified this data."""
-        self._current_module_state = self._send_command("list-modules", True)
-
-    def turn_on(self, sink_name, source_name):
-        """Send a command to pulseaudio to turn on the loopback."""
-        self._send_command(str.format(LOAD_CMD, sink_name, source_name), False)
-
-    def turn_off(self, module_idx):
-        """Send a command to pulseaudio to turn off the loopback."""
-        self._send_command(str.format(UNLOAD_CMD, module_idx), False)
-
-    def get_module_idx(self, sink_name, source_name):
-        """For a sink/source, return its module id in our cache, if found."""
-        result = re.search(
-            str.format(MOD_REGEX, re.escape(sink_name), re.escape(source_name)),
-            self._current_module_state,
-        )
-        if result and result.group(1).isdigit():
-            return int(result.group(1))
-        return -1
-
-
 class PALoopbackSwitch(SwitchEntity):
     """Representation the presence or absence of a PA loopback module."""
 
     def __init__(self, hass, name, pa_server, sink_name, source_name):
         """Initialize the Pulseaudio switch."""
-        self._module_idx = -1
+        self._module_idx = None
         self._hass = hass
         self._name = name
         self._sink_name = sink_name
         self._source_name = source_name
         self._pa_svr = pa_server
 
+        self.update()
+
+    def _get_module_idx(self):
+        try:
+            self._pa_svr.connect()
+
+            for m in self._pa_svr.module_list():
```

Comment:
I think that is not PEP008 conform, use `module`

Reply:
Done.